types/log_parser/logfile.pp

```diff
@@ -0,0 +1,5 @@
+#https://wiki.opnfv.org/display/fastpath/Logparser+plugin+HLD
```

Comment:
The types should have spec tests. Here's an example at https://github.com/sensu/sensu-puppet/blob/master/spec/type_aliases/backend_url_spec.rb

Basically you have two arrays, one of things that should pass and another of things that should fail.

Reply:
Should I use the same approach? Asking since I see it uses shoulda-matchers (If im not mistaken allow_value method is a part of it) Asking since if i'm to use the same approach to testing it I would have to add shoulda-matchers since I dont believe that puppet uses it right now. Or I could go with the approach that is currently example at https://github.com/voxpupuli/puppet-collectd/blob/e1f29e95d3af724eb25b752b16594463c282e7e1/spec/defines/collectd_plugin_snmp_data_spec.rb but that only test if the values are set if im not wrong.